quic/s2n-quic-platform/src/message/msg.rs

```diff
@@ -205,6 +220,94 @@ impl MessageTrait for msghdr {
     }
 }
 
+#[inline]
+pub(super) unsafe fn alloc<T: Copy + Sized, F: Fn(&mut T) -> &mut msghdr>(
```

Comment:
What is unsafe about this? Can we add a SAFETY comment on the function itself?

At minimum based on my reading:

* T can be initialized with zero bytes and be a valid T

Reply:
This code was already merged in #1787. I will open a separate PR for this.

Reply:
https://github.com/aws/s2n-quic/pull/1807